modules/tensor_mechanics/src/interfacekernels/CZMInterfaceKernel.C

```diff
@@ -80,16 +82,16 @@ CZMInterfaceKernel::computeQpJacobian(Moose::DGJacobianType type)
   switch (type)
   {
     case Moose::ElementElement: // Residual_sign -1  ddeltaU_ddisp sign -1;
-      jac *= _test[_i][_qp] * _phi[_j][_qp];
+      jac *= _test[_i][_qp] * _vars[_component]->phiFace()[_j][_qp];
```

Comment:
Could you explain what are the differences?  _phi[_j][_qp] should be the same as _var->phiFace()[_j][_qp] ? and disp_x/y/z would be using same shape function?

Reply:
@jiangwen84 For the diagonal part of the jacobian there is not a difference, I could have left it as it was. I changed it to be consistent with the formulation of the off-diagonal part. I can revert it if you wish.

Reply:
by the way, the variable `_component` is related to the displacement component that kernel is working on.

Reply:
For off-diagonal part, could you explain their difference? I did not quite follow that.

Reply:
The traction in the global coordinate system is a function of the displacement jump in all directions.

What you need in the jacobian is the derivative of the residual wrt to the discrete displacement.

Consider the material already provides dtraction_i/djump_j wrt to the continuous displacement jump then the off-diagonal part of the jacobian is dtraction[component]/djump[offdiag] * djump[offdiag]/du[offdiag][_j].

now because the material provides dtraction[component]/djump[offdiag] in the kernel we must compute djump[offdiag]/du[offdiag][_j]

jump[offdiag] = u_neighbor[offdiag] -  u_master[offdiag]

hence:
1  djump[offdiag]/du[offdiag][_j] (neighbor) = _vars[off_diag]->phiFaceNeighbor[_j]
2  djump[offdiag]/du[offdiag][_j] (master) = -_vars[off_diag]->phiFace[_j]

does this help?

Reply:
I should have added, the traction T[x] = f(u_x,u_y,u_z)

Reply:
My question is why _vars[off_diag]->phiFaceNeighbor[_j] is not equal to _phi_neighbor[_j][_qp]?

Reply:
because `_phi` and `_phi_neighbor` are defined as
`_phi(_assembly.phiFace(_var))` and
`_phi_neighbor(_assembly.phiFaceNeighbor(_neighbor_var)),`

notice that I'm using `_vars` not `_var`

`_phi` `_phi_neighbor` are only associated to the disaplcement component `_component`, see [here](https://github.com/idaholab/moose/blob/984907a4e35c65cfdd84a601581be5e9bd2447bd/framework/src/interfacekernels/InterfaceKernel.C#L48-L59)

Reply:
For this case, the shape function would be the same for every component of displacements, so why to use which component matters? I must be missing some points. 

It should not matter whether `_phi` is associated to disp_x or disp_y. It is basically a shape function on that element surface. Correct?

Reply:
Wait, I believe we are confusing ourselves.
For the diagonal part of the jacobian you are right, and as I wrote I before I can revert the code back and use phi and phi_neighbor if you like it more.
The same is not true for the off diagonal part of the jacobian.
This PR fixes the off-diagonal part of the jacobian.

Reply:
I still cannot understand why that would be different for off diagonal part. Do you have a minute to talk over Teams or whatever you prefer? Thanks.